pkg/v0/cmd.go

```diff
@@ -354,6 +358,14 @@ func cherryPick(ctx context.Context, logger *logrus.Entry, c internal.Commit, co
 		internal.WithEnv(exec.CommandContext(ctx,
 			"make", "generate-manifests",
 		), os.Environ()...),
+	}
+
+	commands := []*exec.Cmd{
+		exec.CommandContext(ctx,
+			"git", append([]string{"add",
+				"vendor"},
```

Comment:
Why do we need to do this if we're explicitly commit it in the next command? Also, is this change related to deferred validation, or not?

Reply:
So, here's the weird thing, without this, there are _many_ unstaged files in the `vendor` directory. It's hard to tell _exactly_ when this occurs, it would require a bunch of extra debug to determine. It's not happening during the `cherry-pick`, I suspect it's one of the `go mod vendor`, but I'm not sure _at this time_ which one...

Reply:
Admittedly, the `add` was a Q&D fix.

Reply:
Ok, the problem is that the files being left over are _untacked_ files. `git commit <directory>` will only commit files that are _tracked_. When `go mod vendor` is run, there may be new files added which are considered _untracked_ and need to be explicitly added. I verified this manually using `git`. So, the `git add` is necessary, but it should only be done for the `vendor` directory, as shown here.

Reply:
@stevekuznetsov See above comment, re: `vendor` directory only.